tests/testsuite/update.rs

```diff
@@ -1489,6 +1489,1381 @@ fn update_precise_yanked_multiple_presence() {
+            "Cargo.toml",
+            r#"
+                # Check if formatting is preserved
+
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                pre  =  "0.1.0"  # Comment
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("pre", "0.2.0-beta").publish();
+    Package::new("pre", "0.2.0").publish();
+
+    p.cargo("update -Zunstable-options -p pre --precise 0.2.0-beta")
+        .masquerade_as_nightly_cargo(&["update-precise-breaking"])
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[ERROR] failed to select a version for the requirement `pre = "^0.1.0"`
+candidate versions found which didn't match: 0.2.0-beta
+location searched: `dummy-registry` index (which is replacing registry `crates-io`)
+required by package `foo v0.0.1 ([ROOT]/foo)`
+if you are looking for the prerelease package it needs to be specified explicitly
+    pre = { version = "0.2.0-beta" }
+perhaps a crate was updated and forgotten to be re-vendored?
+
+"#]])
+        .run();
+
+    let manifest = p.read_file("Cargo.toml");
+    assert_e2e().eq(
+        &manifest,
+        str![[r#"
+
+                # Check if formatting is preserved
+
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                pre  =  "0.1.0"  # Comment
+            
+"#]],
+    );
+}
+
+#[cargo_test]
+fn update_precise_breaking_upgrade_from_pre_release() {
+    Package::new("pre", "1.0.0-alpha").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                # Check if formatting is preserved
+
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                pre  =  "1.0.0-alpha"  # Comment
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("pre", "2.0.0").publish();
+    Package::new("pre", "2.0.1").publish();
+
+    p.cargo("update -Zunstable-options -p pre --precise 2.0.0")
+        .masquerade_as_nightly_cargo(&["update-precise-breaking"])
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[ERROR] failed to select a version for the requirement `pre = "^1.0.0-alpha"`
+candidate versions found which didn't match: 2.0.0
+location searched: `dummy-registry` index (which is replacing registry `crates-io`)
+required by package `foo v0.0.1 ([ROOT]/foo)`
+perhaps a crate was updated and forgotten to be re-vendored?
```

Comment:
Oh no this error message is bad 😢

Reply:
Keep in mind this is without the feature implemented. In the next commit this becomes:

```
[UPGRADING] pre ^1.0.0-alpha -> ^2.0.0
[UPDATING] `dummy-registry` index
[UPDATING] pre v1.0.0-alpha -> v2.0.0
```